plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -777,6 +777,20 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 			}
 		}
 
+		$metadata_fields = array_filter(
+			$request->get_params(),
+			function( $k ) {
+				return str_starts_with( $k, 'meta_data_' );
```

Comment:
Super nitpick: we could use a double underscore here to slim down the very minor chance of conflict with plugins to nearly 0.

```suggestion
				return str_starts_with( $k, 'meta_data__' );
```